### PR TITLE
Use entry points for Custodia plugins

### PR DESCRIPTION
**Pull request by tiran**
Signed-off-by: Christian Heimes <cheimes@redhat.com>

### CODE REVIEW

**Comment by simo5** [comment]
This will break IPA installations.

Here is an example config file:
```
[global]
server_version = "IPAKeys/0.0.1"
server_socket = /run/httpd/ipa-custodia.sock
auditlog = /var/log/ipa-custodia.audit.log

[auth:simple]
handler = custodia.httpd.authenticators.SimpleCredsAuth
uid = 48
gid = 48

[auth:header]
handler = custodia.httpd.authenticators.SimpleHeaderAuth
header = GSS_NAME

[authz:kemkeys]
handler = ipapython.secrets.kem.IPAKEMKeys
paths = /keys
store = ipa
server_keys = /etc/ipa/custodia/server.keys

[store:ipa]
handler = ipapython.secrets.store.iSecStore
ldap_uri = ldapi://%2Fvar%2Frun%2Fslapd-DEV-LAN.socket

[/keys]
handler = custodia.secrets.Secrets
allowed_keytypes = kem
store = ipa
```

As you can see it extends custodia with its own python modules (not under the custodia directory tree), and this patch would not be able to import them.

File: setup.py [hunk 1]
```diff
@@ -46,6 +46,33 @@
         'console_scripts': [
             'custodia = custodia.server:main'
         ],
+        'custodia.authenticators': [
+            'simplecreds = custodia.httpd.authenticators:SimpleCredsAuth',
```
Comment (simo5) [inline setup.py]: why just simplecreds, instead of SimpleCresAuth or at least simplecredsauth ?

Comment (tiran) [inline setup.py]: auth in simplecredsauth is redundant information. The plugin is qualified by its name space.

Comment (simo5) [inline setup.py]: The plugin may be, but lloking at the config file now it is a litte bit less clear.
I think it woiuld help to retain the original names, camel cased, so I was asking why the change, is it just personal aestethic preference ? Or is there some more technical/style/etc reason behind it ?
